src/shared/models/asset/asset.service.ts

```diff
@@ -1,13 +1,14 @@
 import { Injectable } from '@nestjs/common';
+import { Blockchain } from 'src/ain/node/node.service';
 import { AssetRepository } from 'src/shared/models/asset/asset.repository';
 import { Asset } from './asset.entity';
 
 @Injectable()
 export class AssetService {
   constructor(private assetRepo: AssetRepository) {}
 
-  async getAllAsset(): Promise<Asset[]> {
-    return this.assetRepo.find();
+  async getAllAsset(blockchain: Blockchain): Promise<Asset[]> {
+    return this.assetRepo.find({ where: { blockchain } });
```

Comment:
It might be a good idea to deploy this change in a later stage because all logged in users won't see any assets any more.

Reply:
Or we are falling back to DEFICHAIN, if blockchain is not present. In that way we make sure, that nothing changes for already logged in users, and BTC / ETH addresses are getting there respective assets. wdyt?

Reply:
Yes, makes sense 👍